src/components/swapv2/LiquiditySourcesPanel/index.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useState } from 'react'
+import { Flex, Box } from 'rebass'
+import { ArrowLeft } from 'react-feather'
+import styled from 'styled-components'
+import { t } from '@lingui/macro'
+
+import { useActiveWeb3React } from 'hooks'
+import useDebounce from 'hooks/useDebounce'
+import useAggregatorStats from 'hooks/useAggregatorStats'
+import { dexListConfig } from 'constants/dexes'
+
+import SearchBar from './SearchBar'
+
+type Props = {
+  className?: string
+  onBack: () => void
+}
+
+const BackIconWrapper = styled(ArrowLeft)`
+  height: 20px;
+  width: 20px;
+  margin-right: 10px;
+  cursor: pointer;
+  path {
+    stroke: ${({ theme }) => theme.text} !important;
+  }
+`
+
+const BackText = styled.span`
+  font-size: 18px;
+  font-weight: 500;
+  color: ${({ theme }) => theme.text};
+`
+
+const LiquiditySourcesPanel: React.FC<Props> = ({ className, onBack }) => {
+  const { chainId } = useActiveWeb3React()
+  const [searchText, setSearchText] = useState('')
+  const debouncedSearchText = useDebounce(searchText.toLowerCase(), 200)
+
+  const { data, error } = useAggregatorStats(chainId)
+  if (error || !data) {
+    onBack()
+    return null
+  }
+
+  const dexIDs = Object.keys(data.pools)
+  if (dexIDs.length === 0) {
+    onBack()
+    return null
+  }
+
+  const visibleDEXes = dexIDs
+    .map(id => dexListConfig[id])
+    .filter(Boolean)
+    .filter(({ name }) => name.toLowerCase().includes(debouncedSearchText))
+
+  return (
+    <Box width="100%" className={className}>
+      <Flex
+        width={'100%'}
+        flexDirection={'column'}
+        sx={{
+          rowGap: '20px',
+        }}
+      >
+        <Flex
+          alignItems="center"
+          sx={{
+            // this is to make the arrow stay exactly where it stays in Swap panel
+            marginTop: '5px',
+          }}
+        >
+          <BackIconWrapper onClick={onBack}></BackIconWrapper>
+          <BackText>{t`Liquidity Sources`}</BackText>
+        </Flex>
+
+        <SearchBar text={searchText} setText={setSearchText} />
+
+        <div className="sources">
+          {visibleDEXes.map(({ name, icon }) => (
+            <div className="source" key={name}>
+              <div className="imageWrapper">
+                <img src={icon} />
+              </div>
+
+              <span>{name}</span>
+            </div>
+          ))}
+        </div>
+      </Flex>
+    </Box>
+  )
+}
+
+export default styled(LiquiditySourcesPanel)`
+  .sources {
```

Comment:
Dont use class here. Make component you want to refer here a separated styled component.

For example: Dont write `<div className="sources">` and ``export default styled(LiquiditySourcesPanel)` .sources {...}``, write `<WrapperSource>` and ``const WrapperSource = styled.div`...` ``

Reply:
Why?

Reply:
resolve on slack discussion